src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java

```diff
@@ -145,8 +147,8 @@ private void postCreateCaseInCcdProcess(SscsCaseData caseData,
 
     SscsCaseData convertAppealToSscsCaseData(SyaCaseWrapper appeal) {
 
-        String firstHalfOfPostcode = regionalProcessingCenterService
-            .getFirstHalfOfPostcode(appeal.getContactDetails().getPostCode());
+        String postCode = appeal.getContactDetails().getPostCode();
+        String firstHalfOfPostcode = getFirstHalfOfPostcode(postCode);
         RegionalProcessingCenter rpc = regionalProcessingCenterService.getByPostcode(firstHalfOfPostcode);
```

Comment:
Ryan, As you have done some refactoring as well, Isn't it nice to let getByPostcode method to get the first half of the post code? As both the methods are in RegionalProcessingCenterService and assuming getByPostcode would not work differently if you pass the full postcode.

Reply:
That method is in sscs-common, it already does concatenation of the string postcode. It was not in my remit to change it. If I was to I would make it less mad and write lots of unit tests to handle full and partial postcodes. Your welcome to change it, I didn't because it didn't affect populating the venue.

Reply:
Agree with you, I added this comment as you had done some refactoring and in case if you had missed this point. I'll mark this as resolved.